### ISSUE TITLE
Many tests now fail with Invalid_argument("compare: functional value")

### ISSUE THREAD

**Issue by jmid** [ocaml-multicore/multicoretests#348]
Within the last 1-2 days I'm starting to see failures related to comparing functions in tests that haven't triggered such errors before, which haven't been changed recently, and which are not (intentionally!) comparing functions.

Here's an example from Linux 5.1:
https://github.com/ocaml-multicore/multicoretests/actions/runs/4994151783/jobs/8944304277
```
random seed: 495041348
generated error fail pass / total     time test name

[ ]    0    0    0    0 / 1000     0.0s Lin DSL Queue test with Domain
Fatal error: exception Invalid_argument("compare: functional value")
File "src/queue/dune", line 4, characters 7-20:
4 |  (name lin_tests_dsl)
           ^^^^^^^^^^^^^
(cd _build/default/src/queue && ./lin_tests_dsl.exe --verbose)
Command exited with code 2.
```

Observations:
- spotted on 5.1 and `trunk` (we only test 5.0 weekly now) and
- across Linux, macOS, and Windows
- only seen on Lin DSL tests so far
- seen on Queue, Dynlink, Weak, Bigarray, Ephemeron, ... tests

**Comment by jmid** [comment]
It turns out the reason for the failures was an element comparison added to the 0.21 QCheck list shrinker: https://github.com/c-cube/qcheck/pull/280

**Comment by jmid** [comment]
This is fixed with the release of QCheck 0.21.1
